src/main/java/appeng/block/AEBaseSlabBlock.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.block;
+
+
+import java.util.EnumSet;
+import java.util.Random;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockSlab;
+import net.minecraft.client.renderer.texture.IIconRegister;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.IIcon;
+import net.minecraft.util.MovingObjectPosition;
+import net.minecraft.world.World;
+import appeng.core.features.AEFeature;
+import appeng.core.features.IAEFeature;
+import appeng.core.features.IFeatureHandler;
+import appeng.core.features.SlabBlockFeatureHandler;
+
+
+public class AEBaseSlabBlock extends BlockSlab implements IAEFeature
+{
+	private final IFeatureHandler features;
+	public AEBaseBlock block;
+	public int meta;
+	public boolean isDoubleSlab;
+	public AEBaseSlabBlock slabs;
+	public AEBaseSlabBlock dSlabs;
+	public final String name;
+
+	public AEBaseSlabBlock( AEBaseBlock block, int meta, EnumSet<AEFeature> features, boolean isDoubleSlab, String name )
+	{
+        super( isDoubleSlab, block.getMaterial() );
+		this.block = block;
+		this.meta = meta;
+		this.name = name;
+		this.isDoubleSlab = isDoubleSlab;
+		this.setBlockName( "appliedenergistics2." + name );
+        this.setHardness( block.getBlockHardness( null, 0, 0, 0 ) );
+        this.setResistance( block.getExplosionResistance( null ) * 5.0F / 3.0F );
+        this.setStepSound( block.stepSound );
+        this.useNeighborBrightness = true;
+		if (!isDoubleSlab)
+		{
+			this.dSlabs = new AEBaseSlabBlock( block, meta, features, true, name + ".double" ).setSlabs(this);
+			this.slabs = this;
+		} else {
+			this.dSlabs = this;
+		}
+		this.features = isDoubleSlab ? new SlabBlockFeatureHandler( features, this ) : new SlabBlockFeatureHandler( features, this );
+	}
+
+	public AEBaseSlabBlock setSlabs(AEBaseSlabBlock slabs)
+	{
+		this.slabs = slabs;
+		return this;
+	}
+
+	@Override
+	public IFeatureHandler handler()
+	{
+		return this.features;
+	}
+
+	@Override
+	public void postInit()
+	{
+		// Override to do stuff
+	}
+
+	@Override
+	public IIcon getIcon(int dir, int meta)
+	{
+		return block.getIcon( dir, this.meta );
+	}
+
+	@Override
+	public String func_150002_b(int p_150002_1_)
+	{
+		return this.getUnlocalizedName();
+	}
+
+	@Override
+	public void registerBlockIcons(IIconRegister reg) { }
+
+	@Override
+	public Item getItemDropped(int meta, Random rand, int fortune)
+	{
+	    return Item.getItemFromBlock(this.slabs);
+	}
+
+	@Override
+	public int quantityDropped(int meta, int fortune, Random random)
```

Comment:
It does exactly the same as `BlockSlab.quantityDropped()`.

And use the "Add a line note" button, it is easier to keep the context.